src/sections/channels.tsx

```diff
@@ -157,20 +157,28 @@ function ChannelsPage() {
   };
 
   useEffect(() => {
-    const queryParams = new URLSearchParams({
-      apiToken: loginData.apiToken!,
-      apiEndpoint: loginData.apiEndpoint!,
-    }).toString();
+    if (loginData.apiEndpoint && loginData.apiToken) {
+      const queryParams = new URLSearchParams({
+        apiToken: loginData.apiToken,
+        apiEndpoint: loginData.apiEndpoint,
+      }).toString();
+      set_queryParams(queryParams);
+    }
+  }, [loginData.apiToken, loginData.apiEndpoint]);
 
-    set_queryParams(queryParams);
+  useEffect(() => {
+    const currentHash = window.location.hash;
+    const defaultHash =
+      currentHash === '#incoming' || currentHash === '#outgoing'
+        ? currentHash
+        : '#incoming';
 
-    if (queryParams) {
-      console.log(`queryParams: ${queryParams}`);
-      navigate(`?${queryParams}#incoming`, { replace: true });
-    }
+    const defaultTabIndex = defaultHash === '#outgoing' ? 1 : 0;
+    setTabIndex(defaultTabIndex);
+    handleHash(defaultTabIndex);
 
     handleRefresh();
-  }, [loginData.apiToken, loginData.apiEndpoint, navigate]);
+  }, [queryParams]);
```

Comment:
is this dependency needed here?

Reply:
Yes because we need to make sure the queryParams are set before executing `handleHash` which adds the queryParams and hash to the url.

If the dependency is not there, the first time we enter the channels subpage we would get the url without the credentials